instrumentation/google.golang.org/grpc/otelgrpc/config.go

```diff
@@ -257,3 +258,16 @@ func (o spanStartOption) apply(c *config) {
 func WithSpanOptions(opts ...trace.SpanStartOption) Option {
 	return spanStartOption{opts}
 }
+
+type attributesOption struct{ a []attribute.KeyValue }
+
+func (o attributesOption) apply(c *config) {
+	if o.a != nil {
+		c.Attributes = o.a
+	}
+}
+
+// WithAttributes returns an Option to add custom attributes to the spans and metrics.
```

Comment:
I find adding attributes to both spans and metrics problematic. For spans attributes are added to give more context, but for metrics they are used also for aggregation.

Reply:
@dashpole WDYT?

Reply:
right... People should probably use `WithSpanOptions(WithAttributes(...` to add span attributes, right?

Reply:
I am against span options as it gives to much (changing span kind etc). See https://github.com/open-telemetry/opentelemetry-go-contrib/pull/5431. Maybe dedicated options to set attributes for each signal?

Reply:
@dashpole updated